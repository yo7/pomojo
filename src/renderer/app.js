import React from 'react'
import { render } from 'react-dom'

render(
  <div>Init</div>,
  document.getElementById('app')
)
