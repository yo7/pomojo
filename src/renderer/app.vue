<template>
  <timer-board></timer-board>
</template>

<script>
import TimerBoard from './components/timer/timer-board'

export default {
  name: 'app',
  components: {
    'timer-board': TimerBoard
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
