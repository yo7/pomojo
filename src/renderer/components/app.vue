<template>
  <div>{{ msg }}</div>
</template>

<script>
export default {
  name: 'app',
  data() {
    return {
      msg: 'this is a message'
    }
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
