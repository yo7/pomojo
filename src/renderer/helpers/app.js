import {ipcRenderer} from 'electron'

const app = {
  close() {
    ipcRenderer.send('close')
  }
}

export default app
