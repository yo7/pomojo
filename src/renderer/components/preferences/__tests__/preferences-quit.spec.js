import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import app from '../../../helpers/app'
import PreferencesQuit from '../preferences-quit.vue'

describe('Preferences Quit', () => {
  const wrapper = shallow(PreferencesQuit)

  it('close app on clicked', () => {
    jest.spyOn(app, 'close')
    wrapper.trigger('click')
    expect(app.close).toHaveBeenCalled()
  })

  it('renders correctly', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error()
      }
      expect(str).toMatchSnapshot()
    })
  })
})
