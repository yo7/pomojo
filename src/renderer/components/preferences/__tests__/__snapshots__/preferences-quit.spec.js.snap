// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preferences Quit renders correctly 1`] = `
<span class="preferences-quit"><i class="fa fa-power-off icon"></i>
  Quit Pomojo
</span>
`;
